src/Build.UnitTests/Utilities_Tests.cs

```diff
@@ -63,6 +60,23 @@ public void GetTextFromTextNodeWithXmlComment7()
             // Should get XML; note space after x added
             Assert.Equal("<!-- bar; baz; --><!-- bar --><x />", xmlContents);
         }
+
+        [Fact]
+        public void HandshakesDiffer()
+        {
+            int numHandshakeOptions = (int)Math.Pow(2, Enum.GetNames(HandshakeOptions.None.GetType()).Length - 1);
+            Dictionary<long, int> handshakes = new Dictionary<long, int>();
+            for (int i = 0; i < numHandshakeOptions; i++)
+            {
+                long nextKey = CommunicationsUtilities.GetHostHandshake((HandshakeOptions)i);
+                if (handshakes.TryGetValue(nextKey, out int collision))
+                {
+                    CommunicationsUtilities.Trace("There was a collision between {0} and {1}.", collision, i);
```

Comment:
Why is this `CommunicationsUtilities.Trace` instead of an xunit logging trace? Or for that matter just a failed assertion?

Reply:
I can switch to _output.WriteLine. I don't want a failed assertion because if there are multiple collisions, I want to see them all.